src/realm/util/file.cpp

```diff
@@ -1098,7 +1098,10 @@ static void _unlock(int m_fd)
     do {
         r = flock(m_fd, LOCK_UN);
     } while (r != 0 && errno == EINTR);
-    REALM_ASSERT_RELEASE_EX(r == 0 && "File::unlock()", r, errno);
+    if (r && errno) {
+        throw RuntimeError(ErrorCodes::RuntimeError,
```

Comment:
You could make this a SystemError so that we get the errno transferred.

Reply:
Yes, please, SystemError would be better and consistent with other cases. Essentially File throws either FileAccessError or SystemError with errno codes. We should also preserve error code itself - it's very valuable piece of erroneous situation.

Btw, the question is if we can actually recover from such situation because as opposed to assert, exception can be catched and some operation can be retried. Are we safe in this situation after unlock failure?

Reply:
I don't think we can recover @kiburtse. It is likely a limitation in the FS. The idea is to let the SDK to wrap up this error and handle with this on their level.

Reply:
Nicola is correct. There is no way for the SDK to recover from this, so we will just end up throwing an exception to the user.

Reply:
What i meant by that is about later user code. I'd assume that _unlock call happens pretty deep in the call stack on write. Is it safe to use realm after that if user code will catch this exception and ignore? Will it introduce new obscure issues after this exception? Because the original user error about accessing external storage is not what use code should do until implemented properly.